src/lib/sw.js

```diff
@@ -117,29 +93,21 @@ workboxRouting.registerRoute(
 // Configure default cache for standard web.dev files: the offline page, various images etc.
 precacheAndRoute(manifest);
 
-/**
- * Match "/foo-bar/ "and "/foo-bar/as/many/of-these-as-you-like/" (with optional trailing
- * "index.html"), normal page nodes for web.dev. This only matches on pathname.
- */
-const contentPathRe = new RegExp("^/([\\w-]+/)*(|index.html)$");
-
 /**
  * Match "/foo-bar" and "/foo-bar/as-many/but/no/trailing/slash" (but not "/foo/bar/index.html").
  * This only matches on pathname (so it must always start with "/").
  */
 const untrailedContentPathRe = new RegExp("^(/[\\w-]+)+$");
 
 /**
- * Send normal nodes to cache first.
+ * Match fetches for patials, for SPA requests. Matches "/foo-bar/index.json" and
+ * "/foo-bar/many/parts/index.json", for partial SPA requests.
  */
-workboxRouting.registerRoute(
-  ({url, event}) => {
-    return url.host === self.location.host && contentPathRe.test(url.pathname);
-  },
-  new workboxStrategies.NetworkFirst({
-    plugins: [normalizeIndexCacheKeyPlugin],
-  }),
-);
+const partialPathRe = new RegExp("^/([\\w-]+/)*index\\.json$");
+const partialStrategy = new workboxStrategies.NetworkFirst();
```

Comment:
Do you want to add https://developers.google.com/web/tools/workbox/modules/workbox-expiration into the mix here? I'm wary of a content cache that grows unbounded.

Reply:
I agree but I think it's unrelated. I've sent #2270 for you to review. I'll merge the changes here if you're happy.